contracts/interfaces/ITokenHandler.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { ITokenManagerType } from './ITokenManagerType.sol';
+
+/**
+ * @title ITokenManager Interface
+ * @notice This interface is responsible for handling tokens before initiating an interchain token transfer, or after receiving one.
+ */
+interface ITokenHandler is ITokenManagerType {
```

Comment:
Can we avoid inheriting the token manager type? It's not being used directly here, and avoids bytecode change if new types are added there

Reply:
moved ITokenManagerType to the actual TokenHandler contract (whose bytecode needs to change when we add more types because it needs to handle them)